lib/datadog/core/environment/ext.rb

```diff
@@ -4,6 +4,7 @@
 module Datadog
   module Core
     module Environment
+      # @public_api
       module Ext
```

Comment:
Does this need to be public? 🤔 

It seems a bit limiting for this to be a public API, especially because customers can trivially get all of this from the same place we get 'em.

Also applies to other similar classes/modules such as Environment::Identity, Environment::VariableHelpers, etc